presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestBingTileFunctions.java

```diff
@@ -80,6 +82,56 @@ public void testSerialization()
         assertEquals(tile, objectMapper.readerFor(BingTile.class).readValue(json));
     }
 
+    @Test
+    void testBingTileEncoding()
+    {
+        Random random = new Random(1491);
+        for (int zoom = 0; zoom <= MAX_ZOOM_LEVEL; zoom++) {
+            int numIterations = Math.min(1000, (int) Math.pow(4, zoom));
+            for (int iter = 0; iter < numIterations; iter++) {
+                BingTile expected = BingTile.fromCoordinates(
+                        random.nextInt(1 << zoom),
+                        random.nextInt(1 << zoom),
+                        zoom);
+                BingTile actual = BingTile.decode(expected.encode());
+                assertEquals(actual, expected);
+            }
+        }
+    }
+
+    @Test
+    public void testBingTileCast()
+    {
+        Random random = new Random(1812);
+        for (int zoom = 0; zoom <= MAX_ZOOM_LEVEL; zoom++) {
```

Comment:
This test is only supposed to verify the integration of the tile encoding (that is tested in the test above), and the function mechanism. Instead of fuzzing i would recommend adding just a few simple test cases to verify the integration is in place.